From 78cdc602538af6eea6fb6c0020d146e37f7ccf59 Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin.jansa@lge.com>
Date: Wed, 4 Aug 2021 18:55:40 -0700
Subject: [PATCH] MojOsInternal.h: update for glibc-2.34

* as https://sourceware.org/pipermail/libc-alpha/2021-August/129718.html says:
  * The function pthread_yield has been deprecated; programs should use
    the equivalent standard function sched_yield instead.

  it doesn't cause db8 build itself to fail, but some components which use
  MojOsInternal.h (like keymanager) also use -Werror which causes:
  recipe-sysroot/usr/include/mojodb/core/internal/MojOsInternal.h:345:38: error: 'int pthread_yield()' is deprecated: pthread_yield is deprecated, use sched_yield instead [-Werror=deprecated-declarations]
  |   345 |         return (MojErr) pthread_yield();
  |       |                         ~~~~~~~~~~~~~^~

Signed-off-by: Martin Jansa <martin.jansa@lge.com>
---
 inc/core/internal/MojOsInternal.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/inc/core/internal/MojOsInternal.h b/inc/core/internal/MojOsInternal.h
index 60c6105..ce72da8 100644
--- a/inc/core/internal/MojOsInternal.h
+++ b/inc/core/internal/MojOsInternal.h
@@ -342,7 +342,7 @@ inline MojErr MojThreadDetach(MojThreadT thread)
 inline MojErr MojThreadYield()
 {
 #ifdef __USE_GNU
-	return (MojErr) pthread_yield();
+	return (MojErr) sched_yield();
 #else
 	pthread_yield_np();
 	return MojErrNone;
